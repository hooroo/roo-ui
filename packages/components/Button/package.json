{
  "name": "@roo-ui/button",
  "version": "0.1.0",
  "main": "dist/index.js",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "babel src -d dist"
  },
  "peerDependencies": {
    "prop-types": "^15.6.0",
    "react": "^16.2.0",
    "react-dom": "^16.2.0",
    "styled-components": "^3.1.5"
  }
}
