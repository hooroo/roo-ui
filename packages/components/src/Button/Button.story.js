import React from 'react';
import { storiesOf } from '@storybook/react';
import { withDocs } from 'storybook-readme';
import { boolean } from '@storybook/addon-knobs/react';

import Button from '.';
import README from './README.md';

storiesOf('Components|Button', module)
  .addDecorator(withDocs(README))
  .add('default', () => (
    <Button
      primary={boolean('Primary', false)}
      rounded={boolean('Rounded', false)}
      block={boolean('Block', false)}
      disabled={boolean('Disabled', false)}
    >
      Hello world
    </Button>
  ));
