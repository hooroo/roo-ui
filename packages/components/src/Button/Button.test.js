import React from 'react';
import { qantas as theme } from '@roo-ui/themes';
import { shallowWithTheme } from '@roo-ui/test-utils';

import Button from '.';

describe('<Button />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowWithTheme(<Button>Hello world</Button>, theme);
  });

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('primary', () => {
    beforeEach(() => {
      wrapper = shallowWithTheme(<Button primary>Hello world</Button>, theme);
    });

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('rounded', () => {
    beforeEach(() => {
      wrapper = shallowWithTheme(<Button rounded>Hello world</Button>, theme);
    });

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('block', () => {
    beforeEach(() => {
      wrapper = shallowWithTheme(<Button block>Hello world</Button>, theme);
    });

    it('renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
