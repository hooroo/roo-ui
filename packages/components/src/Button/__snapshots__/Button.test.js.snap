// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> block renders correctly 1`] = `
<StyledComponent
  blacklist={
    Array [
      "m",
      "mt",
      "mr",
      "mb",
      "ml",
      "mx",
      "my",
      "p",
      "pt",
      "pr",
      "pb",
      "pl",
      "px",
      "py",
      "color",
      "bg",
      "boxShadow",
      "rounded",
      "block",
    ]
  }
  block={true}
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "Button-hYXUXp",
        "isStatic": false,
        "lastClassName": "cvQuZd",
        "rules": Array [
          "display:inline-block;margin:0;padding:",
          [Function],
          " ",
          [Function],
          ";font-size:",
          [Function],
          ";font-weight:",
          [Function],
          ";letter-spacing:",
          [Function],
          ";line-height:",
          [Function],
          ";text-align:center;text-decoration:none;text-transform:uppercase;border:",
          [Function],
          ";border-color:transparent;border-radius:",
          [Function],
          ";outline:0;transition:background-color ",
          [Function],
          ";cursor:pointer;appearance:none;",
          [Function],
          " ",
          [Function],
          " ",
          [Function],
          " ",
          [Function],
          " &:hover:not(:disabled){background-color:",
          [Function],
          ";}&:focus{box-shadow:",
          [Function],
          ";}&:disabled{opacity:",
          [Function],
          ";cursor:not-allowed;}",
          [Function],
          " ",
          [Function],
        ],
      },
      "defaultProps": Object {
        "blacklist": Array [
          "m",
          "mt",
          "mr",
          "mb",
          "ml",
          "mx",
          "my",
          "p",
          "pt",
          "pr",
          "pb",
          "pl",
          "px",
          "py",
          "color",
          "bg",
          "boxShadow",
          "rounded",
          "block",
        ],
        "variant": "default",
      },
      "displayName": "Button",
      "foldedComponentIds": Array [],
      "propTypes": Object {
        "bg": [Function],
        "block": [Function],
        "boxShadow": [Function],
        "color": [Function],
        "m": [Function],
        "mb": [Function],
        "ml": [Function],
        "mr": [Function],
        "mt": [Function],
        "mx": [Function],
        "my": [Function],
        "p": [Function],
        "pb": [Function],
        "pl": [Function],
        "pr": [Function],
        "pt": [Function],
        "px": [Function],
        "py": [Function],
        "rounded": [Function],
      },
      "render": [Function],
      "styledComponentId": "Button-hYXUXp",
      "target": Object {
        "$$typeof": Symbol(react.forward_ref),
        "displayName": "Clean.button",
        "render": [Function],
      },
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  theme={
    Object {
      "alertStyles": Object {
        "default": Object {
          "bg": "greys.porcelain",
        },
        "error": Object {
          "bg": "ui.errorBackground",
          "icon": Object {
            "color": "ui.error",
            "name": "warning",
          },
        },
        "info": Object {
          "bg": "ui.infoBackground",
          "icon": Object {
            "color": "greys.charcoal",
            "name": "info",
          },
        },
        "success": Object {
          "bg": "ui.successBackground",
          "icon": Object {
            "color": "ui.success",
            "name": "checkCircle",
          },
        },
      },
      "borders": Array [
        0,
        "1px solid",
        "2px solid",
        "3px solid",
      ],
      "breakpoints": Array [
        "40rem",
        "52rem",
        "64rem",
      ],
      "buttons": Object {
        "default": Object {
          "backgroundColor": "#323232",
          "color": "white",
        },
        "primary": Object {
          "backgroundColor": "#E40000",
          "color": "white",
        },
      },
      "colors": Object {
        "blue": "#8DE2E0",
        "brand": Object {
          "primary": "#E40000",
          "quaternary": "#F9F3E9",
          "secondary": "#8DE2E0",
          "tertiary": "#D20000",
        },
        "brightSun": "#FBCB3B",
        "green": "#35A509",
        "greys": Object {
          "alto": "#DADADA",
          "charcoal": "#323232",
          "dusty": "#999999",
          "mineShaft": "#222222",
          "porcelain": "#F4F5F6",
          "steel": "#666666",
        },
        "lightBlue": "#E7F7F7",
        "lightGreen": "#DEF2DE",
        "lightOrange": "#FCEBCD",
        "maroon": "#870000",
        "orange": "#ED710B",
        "red": "#E00E00",
        "ui": Object {
          "error": "#ED710B",
          "errorBackground": "#FCEBCD",
          "info": "#8DE2E0",
          "infoBackground": "#E7F7F7",
          "link": "#E00E00",
          "linkHover": "#870000",
          "success": "#35A509",
          "successBackground": "#DEF2DE",
        },
        "white": "#FFFFFF",
      },
      "fontFamily": "Ciutadella, sans-serif",
      "fontSizes": Object {
        "2xl": "1.625rem",
        "3xl": "2.25rem",
        "4xl": "2.875rem",
        "base": "1rem",
        "lg": "1.125rem",
        "sm": "0.9375rem",
        "xl": "1.375rem",
        "xs": "0.8125rem",
      },
      "fontWeights": Object {
        "bold": 600,
        "normal": 500,
      },
      "gutters": Object {
        "default": 3,
      },
      "letterSpacings": Object {
        "normal": "0",
        "wide": "0.125em",
      },
      "lineHeights": Object {
        "loose": 1.5,
        "normal": 1.35,
        "tight": 1,
      },
      "maxWidths": Object {
        "default": "80rem",
      },
      "mediaQueries": Array [
        "@media screen and (min-width: 40rem)",
        "@media screen and (min-width: 52rem)",
        "@media screen and (min-width: 64rem)",
      ],
      "opacity": Object {
        "disabled": 0.7,
      },
      "radii": Object {
        "default": "4px",
        "none": 0,
        "rounded": "1000px",
      },
      "shadows": Object {
        "default": "0 1px 1px 0 rgba(0, 0, 0, 0.1)",
        "focus": "0 0 2px 2px #999999;",
        "heavy": "0 2px 2px 0 rgba(0, 0, 0, 0.1)",
        "none": 0,
      },
      "space": Array [
        "0rem",
        "0.25rem",
        "0.5rem",
        "0.75rem",
        "1rem",
        "1.25rem",
        "1.5rem",
        "1.75rem",
        "2rem",
        "2.25rem",
        "2.5rem",
        "2.75rem",
        "3rem",
        "3.25rem",
        "3.5rem",
        "3.75rem",
        "4rem",
        "4.25rem",
        "4.5rem",
        "4.75rem",
        "5rem",
        "5.25rem",
        "5.5rem",
        "5.75rem",
        "6rem",
        "6.25rem",
        "6.5rem",
        "6.75rem",
        "7rem",
        "7.25rem",
        "7.5rem",
        "7.75rem",
        "8rem",
      ],
      "textStyles": Object {
        "caps": Object {
          "fontWeight": 600,
          "textTransform": "uppercase",
        },
        "h1": Object {
          "fontSize": "2.25rem",
          "fontWeight": 600,
          "lineHeight": 1.35,
        },
        "h2": Object {
          "fontSize": "1.625rem",
          "fontWeight": 600,
          "lineHeight": 1.35,
        },
        "h3": Object {
          "fontSize": "1.375rem",
          "fontWeight": 600,
          "lineHeight": 1.5,
        },
        "h4": Object {
          "fontSize": "1.125rem",
          "fontWeight": 600,
          "lineHeight": 1.5,
        },
        "h5": Object {
          "fontSize": "1rem",
          "fontWeight": 600,
          "lineHeight": 1.5,
        },
        "h6": Object {
          "fontSize": "0.9375rem",
          "fontWeight": 600,
          "lineHeight": 1.5,
        },
        "label": Object {
          "color": "#323232",
          "fontSize": "0.9375rem",
          "fontWeight": 600,
        },
        "text": Object {
          "fontSize": "1rem",
          "fontWeight": 500,
          "lineHeight": 1.5,
        },
      },
      "transitions": Object {
        "default": "200ms ease-in",
      },
    }
  }
  variant="default"
>
  Hello world
</StyledComponent>
`;

exports[`<Button /> primary renders correctly 1`] = `
<StyledComponent
  blacklist={
    Array [
      "m",
      "mt",
      "mr",
      "mb",
      "ml",
      "mx",
      "my",
      "p",
      "pt",
      "pr",
      "pb",
      "pl",
      "px",
      "py",
      "color",
      "bg",
      "boxShadow",
      "rounded",
      "block",
    ]
  }
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "Button-hYXUXp",
        "isStatic": false,
        "lastClassName": "cvQuZd",
        "rules": Array [
          "display:inline-block;margin:0;padding:",
          [Function],
          " ",
          [Function],
          ";font-size:",
          [Function],
          ";font-weight:",
          [Function],
          ";letter-spacing:",
          [Function],
          ";line-height:",
          [Function],
          ";text-align:center;text-decoration:none;text-transform:uppercase;border:",
          [Function],
          ";border-color:transparent;border-radius:",
          [Function],
          ";outline:0;transition:background-color ",
          [Function],
          ";cursor:pointer;appearance:none;",
          [Function],
          " ",
          [Function],
          " ",
          [Function],
          " ",
          [Function],
          " &:hover:not(:disabled){background-color:",
          [Function],
          ";}&:focus{box-shadow:",
          [Function],
          ";}&:disabled{opacity:",
          [Function],
          ";cursor:not-allowed;}",
          [Function],
          " ",
          [Function],
        ],
      },
      "defaultProps": Object {
        "blacklist": Array [
          "m",
          "mt",
          "mr",
          "mb",
          "ml",
          "mx",
          "my",
          "p",
          "pt",
          "pr",
          "pb",
          "pl",
          "px",
          "py",
          "color",
          "bg",
          "boxShadow",
          "rounded",
          "block",
        ],
        "variant": "default",
      },
      "displayName": "Button",
      "foldedComponentIds": Array [],
      "propTypes": Object {
        "bg": [Function],
        "block": [Function],
        "boxShadow": [Function],
        "color": [Function],
        "m": [Function],
        "mb": [Function],
        "ml": [Function],
        "mr": [Function],
        "mt": [Function],
        "mx": [Function],
        "my": [Function],
        "p": [Function],
        "pb": [Function],
        "pl": [Function],
        "pr": [Function],
        "pt": [Function],
        "px": [Function],
        "py": [Function],
        "rounded": [Function],
      },
      "render": [Function],
      "styledComponentId": "Button-hYXUXp",
      "target": Object {
        "$$typeof": Symbol(react.forward_ref),
        "displayName": "Clean.button",
        "render": [Function],
      },
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  theme={
    Object {
      "alertStyles": Object {
        "default": Object {
          "bg": "greys.porcelain",
        },
        "error": Object {
          "bg": "ui.errorBackground",
          "icon": Object {
            "color": "ui.error",
            "name": "warning",
          },
        },
        "info": Object {
          "bg": "ui.infoBackground",
          "icon": Object {
            "color": "greys.charcoal",
            "name": "info",
          },
        },
        "success": Object {
          "bg": "ui.successBackground",
          "icon": Object {
            "color": "ui.success",
            "name": "checkCircle",
          },
        },
      },
      "borders": Array [
        0,
        "1px solid",
        "2px solid",
        "3px solid",
      ],
      "breakpoints": Array [
        "40rem",
        "52rem",
        "64rem",
      ],
      "buttons": Object {
        "default": Object {
          "backgroundColor": "#323232",
          "color": "white",
        },
        "primary": Object {
          "backgroundColor": "#E40000",
          "color": "white",
        },
      },
      "colors": Object {
        "blue": "#8DE2E0",
        "brand": Object {
          "primary": "#E40000",
          "quaternary": "#F9F3E9",
          "secondary": "#8DE2E0",
          "tertiary": "#D20000",
        },
        "brightSun": "#FBCB3B",
        "green": "#35A509",
        "greys": Object {
          "alto": "#DADADA",
          "charcoal": "#323232",
          "dusty": "#999999",
          "mineShaft": "#222222",
          "porcelain": "#F4F5F6",
          "steel": "#666666",
        },
        "lightBlue": "#E7F7F7",
        "lightGreen": "#DEF2DE",
        "lightOrange": "#FCEBCD",
        "maroon": "#870000",
        "orange": "#ED710B",
        "red": "#E00E00",
        "ui": Object {
          "error": "#ED710B",
          "errorBackground": "#FCEBCD",
          "info": "#8DE2E0",
          "infoBackground": "#E7F7F7",
          "link": "#E00E00",
          "linkHover": "#870000",
          "success": "#35A509",
          "successBackground": "#DEF2DE",
        },
        "white": "#FFFFFF",
      },
      "fontFamily": "Ciutadella, sans-serif",
      "fontSizes": Object {
        "2xl": "1.625rem",
        "3xl": "2.25rem",
        "4xl": "2.875rem",
        "base": "1rem",
        "lg": "1.125rem",
        "sm": "0.9375rem",
        "xl": "1.375rem",
        "xs": "0.8125rem",
      },
      "fontWeights": Object {
        "bold": 600,
        "normal": 500,
      },
      "gutters": Object {
        "default": 3,
      },
      "letterSpacings": Object {
        "normal": "0",
        "wide": "0.125em",
      },
      "lineHeights": Object {
        "loose": 1.5,
        "normal": 1.35,
        "tight": 1,
      },
      "maxWidths": Object {
        "default": "80rem",
      },
      "mediaQueries": Array [
        "@media screen and (min-width: 40rem)",
        "@media screen and (min-width: 52rem)",
        "@media screen and (min-width: 64rem)",
      ],
      "opacity": Object {
        "disabled": 0.7,
      },
      "radii": Object {
        "default": "4px",
        "none": 0,
        "rounded": "1000px",
      },
      "shadows": Object {
        "default": "0 1px 1px 0 rgba(0, 0, 0, 0.1)",
        "focus": "0 0 2px 2px #999999;",
        "heavy": "0 2px 2px 0 rgba(0, 0, 0, 0.1)",
        "none": 0,
      },
      "space": Array [
        "0rem",
        "0.25rem",
        "0.5rem",
        "0.75rem",
        "1rem",
        "1.25rem",
        "1.5rem",
        "1.75rem",
        "2rem",
        "2.25rem",
        "2.5rem",
        "2.75rem",
        "3rem",
        "3.25rem",
        "3.5rem",
        "3.75rem",
        "4rem",
        "4.25rem",
        "4.5rem",
        "4.75rem",
        "5rem",
        "5.25rem",
        "5.5rem",
        "5.75rem",
        "6rem",
        "6.25rem",
        "6.5rem",
        "6.75rem",
        "7rem",
        "7.25rem",
        "7.5rem",
        "7.75rem",
        "8rem",
      ],
      "textStyles": Object {
        "caps": Object {
          "fontWeight": 600,
          "textTransform": "uppercase",
        },
        "h1": Object {
          "fontSize": "2.25rem",
          "fontWeight": 600,
          "lineHeight": 1.35,
        },
        "h2": Object {
          "fontSize": "1.625rem",
          "fontWeight": 600,
          "lineHeight": 1.35,
        },
        "h3": Object {
          "fontSize": "1.375rem",
          "fontWeight": 600,
          "lineHeight": 1.5,
        },
        "h4": Object {
          "fontSize": "1.125rem",
          "fontWeight": 600,
          "lineHeight": 1.5,
        },
        "h5": Object {
          "fontSize": "1rem",
          "fontWeight": 600,
          "lineHeight": 1.5,
        },
        "h6": Object {
          "fontSize": "0.9375rem",
          "fontWeight": 600,
          "lineHeight": 1.5,
        },
        "label": Object {
          "color": "#323232",
          "fontSize": "0.9375rem",
          "fontWeight": 600,
        },
        "text": Object {
          "fontSize": "1rem",
          "fontWeight": 500,
          "lineHeight": 1.5,
        },
      },
      "transitions": Object {
        "default": "200ms ease-in",
      },
    }
  }
  variant="primary"
>
  Hello world
</StyledComponent>
`;

exports[`<Button /> renders correctly 1`] = `
<StyledComponent
  blacklist={
    Array [
      "m",
      "mt",
      "mr",
      "mb",
      "ml",
      "mx",
      "my",
      "p",
      "pt",
      "pr",
      "pb",
      "pl",
      "px",
      "py",
      "color",
      "bg",
      "boxShadow",
      "rounded",
      "block",
    ]
  }
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "Button-hYXUXp",
        "isStatic": false,
        "rules": Array [
          "display:inline-block;margin:0;padding:",
          [Function],
          " ",
          [Function],
          ";font-size:",
          [Function],
          ";font-weight:",
          [Function],
          ";letter-spacing:",
          [Function],
          ";line-height:",
          [Function],
          ";text-align:center;text-decoration:none;text-transform:uppercase;border:",
          [Function],
          ";border-color:transparent;border-radius:",
          [Function],
          ";outline:0;transition:background-color ",
          [Function],
          ";cursor:pointer;appearance:none;",
          [Function],
          " ",
          [Function],
          " ",
          [Function],
          " ",
          [Function],
          " &:hover:not(:disabled){background-color:",
          [Function],
          ";}&:focus{box-shadow:",
          [Function],
          ";}&:disabled{opacity:",
          [Function],
          ";cursor:not-allowed;}",
          [Function],
          " ",
          [Function],
        ],
      },
      "defaultProps": Object {
        "blacklist": Array [
          "m",
          "mt",
          "mr",
          "mb",
          "ml",
          "mx",
          "my",
          "p",
          "pt",
          "pr",
          "pb",
          "pl",
          "px",
          "py",
          "color",
          "bg",
          "boxShadow",
          "rounded",
          "block",
        ],
        "variant": "default",
      },
      "displayName": "Button",
      "foldedComponentIds": Array [],
      "propTypes": Object {
        "bg": [Function],
        "block": [Function],
        "boxShadow": [Function],
        "color": [Function],
        "m": [Function],
        "mb": [Function],
        "ml": [Function],
        "mr": [Function],
        "mt": [Function],
        "mx": [Function],
        "my": [Function],
        "p": [Function],
        "pb": [Function],
        "pl": [Function],
        "pr": [Function],
        "pt": [Function],
        "px": [Function],
        "py": [Function],
        "rounded": [Function],
      },
      "render": [Function],
      "styledComponentId": "Button-hYXUXp",
      "target": Object {
        "$$typeof": Symbol(react.forward_ref),
        "displayName": "Clean.button",
        "render": [Function],
      },
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  theme={
    Object {
      "alertStyles": Object {
        "default": Object {
          "bg": "greys.porcelain",
        },
        "error": Object {
          "bg": "ui.errorBackground",
          "icon": Object {
            "color": "ui.error",
            "name": "warning",
          },
        },
        "info": Object {
          "bg": "ui.infoBackground",
          "icon": Object {
            "color": "greys.charcoal",
            "name": "info",
          },
        },
        "success": Object {
          "bg": "ui.successBackground",
          "icon": Object {
            "color": "ui.success",
            "name": "checkCircle",
          },
        },
      },
      "borders": Array [
        0,
        "1px solid",
        "2px solid",
        "3px solid",
      ],
      "breakpoints": Array [
        "40rem",
        "52rem",
        "64rem",
      ],
      "buttons": Object {
        "default": Object {
          "backgroundColor": "#323232",
          "color": "white",
        },
        "primary": Object {
          "backgroundColor": "#E40000",
          "color": "white",
        },
      },
      "colors": Object {
        "blue": "#8DE2E0",
        "brand": Object {
          "primary": "#E40000",
          "quaternary": "#F9F3E9",
          "secondary": "#8DE2E0",
          "tertiary": "#D20000",
        },
        "brightSun": "#FBCB3B",
        "green": "#35A509",
        "greys": Object {
          "alto": "#DADADA",
          "charcoal": "#323232",
          "dusty": "#999999",
          "mineShaft": "#222222",
          "porcelain": "#F4F5F6",
          "steel": "#666666",
        },
        "lightBlue": "#E7F7F7",
        "lightGreen": "#DEF2DE",
        "lightOrange": "#FCEBCD",
        "maroon": "#870000",
        "orange": "#ED710B",
        "red": "#E00E00",
        "ui": Object {
          "error": "#ED710B",
          "errorBackground": "#FCEBCD",
          "info": "#8DE2E0",
          "infoBackground": "#E7F7F7",
          "link": "#E00E00",
          "linkHover": "#870000",
          "success": "#35A509",
          "successBackground": "#DEF2DE",
        },
        "white": "#FFFFFF",
      },
      "fontFamily": "Ciutadella, sans-serif",
      "fontSizes": Object {
        "2xl": "1.625rem",
        "3xl": "2.25rem",
        "4xl": "2.875rem",
        "base": "1rem",
        "lg": "1.125rem",
        "sm": "0.9375rem",
        "xl": "1.375rem",
        "xs": "0.8125rem",
      },
      "fontWeights": Object {
        "bold": 600,
        "normal": 500,
      },
      "gutters": Object {
        "default": 3,
      },
      "letterSpacings": Object {
        "normal": "0",
        "wide": "0.125em",
      },
      "lineHeights": Object {
        "loose": 1.5,
        "normal": 1.35,
        "tight": 1,
      },
      "maxWidths": Object {
        "default": "80rem",
      },
      "mediaQueries": Array [
        "@media screen and (min-width: 40rem)",
        "@media screen and (min-width: 52rem)",
        "@media screen and (min-width: 64rem)",
      ],
      "opacity": Object {
        "disabled": 0.7,
      },
      "radii": Object {
        "default": "4px",
        "none": 0,
        "rounded": "1000px",
      },
      "shadows": Object {
        "default": "0 1px 1px 0 rgba(0, 0, 0, 0.1)",
        "focus": "0 0 2px 2px #999999;",
        "heavy": "0 2px 2px 0 rgba(0, 0, 0, 0.1)",
        "none": 0,
      },
      "space": Array [
        "0rem",
        "0.25rem",
        "0.5rem",
        "0.75rem",
        "1rem",
        "1.25rem",
        "1.5rem",
        "1.75rem",
        "2rem",
        "2.25rem",
        "2.5rem",
        "2.75rem",
        "3rem",
        "3.25rem",
        "3.5rem",
        "3.75rem",
        "4rem",
        "4.25rem",
        "4.5rem",
        "4.75rem",
        "5rem",
        "5.25rem",
        "5.5rem",
        "5.75rem",
        "6rem",
        "6.25rem",
        "6.5rem",
        "6.75rem",
        "7rem",
        "7.25rem",
        "7.5rem",
        "7.75rem",
        "8rem",
      ],
      "textStyles": Object {
        "caps": Object {
          "fontWeight": 600,
          "textTransform": "uppercase",
        },
        "h1": Object {
          "fontSize": "2.25rem",
          "fontWeight": 600,
          "lineHeight": 1.35,
        },
        "h2": Object {
          "fontSize": "1.625rem",
          "fontWeight": 600,
          "lineHeight": 1.35,
        },
        "h3": Object {
          "fontSize": "1.375rem",
          "fontWeight": 600,
          "lineHeight": 1.5,
        },
        "h4": Object {
          "fontSize": "1.125rem",
          "fontWeight": 600,
          "lineHeight": 1.5,
        },
        "h5": Object {
          "fontSize": "1rem",
          "fontWeight": 600,
          "lineHeight": 1.5,
        },
        "h6": Object {
          "fontSize": "0.9375rem",
          "fontWeight": 600,
          "lineHeight": 1.5,
        },
        "label": Object {
          "color": "#323232",
          "fontSize": "0.9375rem",
          "fontWeight": 600,
        },
        "text": Object {
          "fontSize": "1rem",
          "fontWeight": 500,
          "lineHeight": 1.5,
        },
      },
      "transitions": Object {
        "default": "200ms ease-in",
      },
    }
  }
  variant="default"
>
  Hello world
</StyledComponent>
`;

exports[`<Button /> rounded renders correctly 1`] = `
<StyledComponent
  blacklist={
    Array [
      "m",
      "mt",
      "mr",
      "mb",
      "ml",
      "mx",
      "my",
      "p",
      "pt",
      "pr",
      "pb",
      "pl",
      "px",
      "py",
      "color",
      "bg",
      "boxShadow",
      "rounded",
      "block",
    ]
  }
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "Button-hYXUXp",
        "isStatic": false,
        "lastClassName": "cvQuZd",
        "rules": Array [
          "display:inline-block;margin:0;padding:",
          [Function],
          " ",
          [Function],
          ";font-size:",
          [Function],
          ";font-weight:",
          [Function],
          ";letter-spacing:",
          [Function],
          ";line-height:",
          [Function],
          ";text-align:center;text-decoration:none;text-transform:uppercase;border:",
          [Function],
          ";border-color:transparent;border-radius:",
          [Function],
          ";outline:0;transition:background-color ",
          [Function],
          ";cursor:pointer;appearance:none;",
          [Function],
          " ",
          [Function],
          " ",
          [Function],
          " ",
          [Function],
          " &:hover:not(:disabled){background-color:",
          [Function],
          ";}&:focus{box-shadow:",
          [Function],
          ";}&:disabled{opacity:",
          [Function],
          ";cursor:not-allowed;}",
          [Function],
          " ",
          [Function],
        ],
      },
      "defaultProps": Object {
        "blacklist": Array [
          "m",
          "mt",
          "mr",
          "mb",
          "ml",
          "mx",
          "my",
          "p",
          "pt",
          "pr",
          "pb",
          "pl",
          "px",
          "py",
          "color",
          "bg",
          "boxShadow",
          "rounded",
          "block",
        ],
        "variant": "default",
      },
      "displayName": "Button",
      "foldedComponentIds": Array [],
      "propTypes": Object {
        "bg": [Function],
        "block": [Function],
        "boxShadow": [Function],
        "color": [Function],
        "m": [Function],
        "mb": [Function],
        "ml": [Function],
        "mr": [Function],
        "mt": [Function],
        "mx": [Function],
        "my": [Function],
        "p": [Function],
        "pb": [Function],
        "pl": [Function],
        "pr": [Function],
        "pt": [Function],
        "px": [Function],
        "py": [Function],
        "rounded": [Function],
      },
      "render": [Function],
      "styledComponentId": "Button-hYXUXp",
      "target": Object {
        "$$typeof": Symbol(react.forward_ref),
        "displayName": "Clean.button",
        "render": [Function],
      },
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  rounded={true}
  theme={
    Object {
      "alertStyles": Object {
        "default": Object {
          "bg": "greys.porcelain",
        },
        "error": Object {
          "bg": "ui.errorBackground",
          "icon": Object {
            "color": "ui.error",
            "name": "warning",
          },
        },
        "info": Object {
          "bg": "ui.infoBackground",
          "icon": Object {
            "color": "greys.charcoal",
            "name": "info",
          },
        },
        "success": Object {
          "bg": "ui.successBackground",
          "icon": Object {
            "color": "ui.success",
            "name": "checkCircle",
          },
        },
      },
      "borders": Array [
        0,
        "1px solid",
        "2px solid",
        "3px solid",
      ],
      "breakpoints": Array [
        "40rem",
        "52rem",
        "64rem",
      ],
      "buttons": Object {
        "default": Object {
          "backgroundColor": "#323232",
          "color": "white",
        },
        "primary": Object {
          "backgroundColor": "#E40000",
          "color": "white",
        },
      },
      "colors": Object {
        "blue": "#8DE2E0",
        "brand": Object {
          "primary": "#E40000",
          "quaternary": "#F9F3E9",
          "secondary": "#8DE2E0",
          "tertiary": "#D20000",
        },
        "brightSun": "#FBCB3B",
        "green": "#35A509",
        "greys": Object {
          "alto": "#DADADA",
          "charcoal": "#323232",
          "dusty": "#999999",
          "mineShaft": "#222222",
          "porcelain": "#F4F5F6",
          "steel": "#666666",
        },
        "lightBlue": "#E7F7F7",
        "lightGreen": "#DEF2DE",
        "lightOrange": "#FCEBCD",
        "maroon": "#870000",
        "orange": "#ED710B",
        "red": "#E00E00",
        "ui": Object {
          "error": "#ED710B",
          "errorBackground": "#FCEBCD",
          "info": "#8DE2E0",
          "infoBackground": "#E7F7F7",
          "link": "#E00E00",
          "linkHover": "#870000",
          "success": "#35A509",
          "successBackground": "#DEF2DE",
        },
        "white": "#FFFFFF",
      },
      "fontFamily": "Ciutadella, sans-serif",
      "fontSizes": Object {
        "2xl": "1.625rem",
        "3xl": "2.25rem",
        "4xl": "2.875rem",
        "base": "1rem",
        "lg": "1.125rem",
        "sm": "0.9375rem",
        "xl": "1.375rem",
        "xs": "0.8125rem",
      },
      "fontWeights": Object {
        "bold": 600,
        "normal": 500,
      },
      "gutters": Object {
        "default": 3,
      },
      "letterSpacings": Object {
        "normal": "0",
        "wide": "0.125em",
      },
      "lineHeights": Object {
        "loose": 1.5,
        "normal": 1.35,
        "tight": 1,
      },
      "maxWidths": Object {
        "default": "80rem",
      },
      "mediaQueries": Array [
        "@media screen and (min-width: 40rem)",
        "@media screen and (min-width: 52rem)",
        "@media screen and (min-width: 64rem)",
      ],
      "opacity": Object {
        "disabled": 0.7,
      },
      "radii": Object {
        "default": "4px",
        "none": 0,
        "rounded": "1000px",
      },
      "shadows": Object {
        "default": "0 1px 1px 0 rgba(0, 0, 0, 0.1)",
        "focus": "0 0 2px 2px #999999;",
        "heavy": "0 2px 2px 0 rgba(0, 0, 0, 0.1)",
        "none": 0,
      },
      "space": Array [
        "0rem",
        "0.25rem",
        "0.5rem",
        "0.75rem",
        "1rem",
        "1.25rem",
        "1.5rem",
        "1.75rem",
        "2rem",
        "2.25rem",
        "2.5rem",
        "2.75rem",
        "3rem",
        "3.25rem",
        "3.5rem",
        "3.75rem",
        "4rem",
        "4.25rem",
        "4.5rem",
        "4.75rem",
        "5rem",
        "5.25rem",
        "5.5rem",
        "5.75rem",
        "6rem",
        "6.25rem",
        "6.5rem",
        "6.75rem",
        "7rem",
        "7.25rem",
        "7.5rem",
        "7.75rem",
        "8rem",
      ],
      "textStyles": Object {
        "caps": Object {
          "fontWeight": 600,
          "textTransform": "uppercase",
        },
        "h1": Object {
          "fontSize": "2.25rem",
          "fontWeight": 600,
          "lineHeight": 1.35,
        },
        "h2": Object {
          "fontSize": "1.625rem",
          "fontWeight": 600,
          "lineHeight": 1.35,
        },
        "h3": Object {
          "fontSize": "1.375rem",
          "fontWeight": 600,
          "lineHeight": 1.5,
        },
        "h4": Object {
          "fontSize": "1.125rem",
          "fontWeight": 600,
          "lineHeight": 1.5,
        },
        "h5": Object {
          "fontSize": "1rem",
          "fontWeight": 600,
          "lineHeight": 1.5,
        },
        "h6": Object {
          "fontSize": "0.9375rem",
          "fontWeight": 600,
          "lineHeight": 1.5,
        },
        "label": Object {
          "color": "#323232",
          "fontSize": "0.9375rem",
          "fontWeight": 600,
        },
        "text": Object {
          "fontSize": "1rem",
          "fontWeight": 500,
          "lineHeight": 1.5,
        },
      },
      "transitions": Object {
        "default": "200ms ease-in",
      },
    }
  }
  variant="default"
>
  Hello world
</StyledComponent>
`;
